sdk/monitor/Azure.Monitor.Query/src/Models/MonitorQueryModelFactory.cs

```diff
@@ -36,5 +36,35 @@ public static LogsQueryResult LogsQueryResult(IReadOnlyList<LogsTable> allTables
             JsonElement errorJson = error.ToObjectFromJson<JsonElement>();
             return new LogsQueryResult(allTables, statisticsJson, visualizationJson, errorJson);
         }
+
+        /// <summary> Initializes a new instance of LogsTableRow. </summary>
+        /// <returns> A new <see cref="Models.LogsTableColumn"/> instance for mocking. </returns>
+        public static LogsTableRow LogsTableRow(IReadOnlyList<LogsTableColumn> columns, Object[] rows)
+        {
+            Dictionary<string, int> columnDictionary = new();
+
+            for (var index = 0; index < columns.Count; index++)
+            {
+                columnDictionary[columns[index].Name] = index;
+            }
+
+            JsonElement row = JsonElementFromObject(rows);
+            return new LogsTableRow(columnDictionary, columns, row);
```

Comment:
Nit - should we move this logic into the `LogsTableRow` class?  If the way the columns mapping works is an implementation detail of that class, it might be cleaner to keep all of it together so any future changes to `LogsTableRow` don't accidentally forget about and break this code.

Reply:
Good idea, I'll make a helper method in LogsTableRow class